internal/corazawaf/transaction.go

```diff
@@ -505,9 +505,12 @@ func (tx *Transaction) MatchRule(r *Rule, mds []types.MatchData) {
 		}
 	}
 
-	tx.matchedRules = append(tx.matchedRules, mr)
-	if tx.WAF.ErrorLogCb != nil && r.Log {
-		tx.WAF.ErrorLogCb(mr)
+	if r.Log {
+		// Only rules with log data will added to the matchedRules list
+		tx.matchedRules = append(tx.matchedRules, mr)
```

Comment:
Main change

Reply:
Should we not return from the top ... if r.Log is not there why to do even some processing (a debug log will help for those cases where a rule has an action but log -- possibly will be atypo only :))   -- cannot see anything obvious breaking here if we exit right from start